packages/aws-cdk-lib/aws-dynamodb/lib/global-table.ts

```diff
@@ -0,0 +1,993 @@
+  private readonly _targetUtilizationPercent?: number;
+
+  public get units() {
+    if (this._units === undefined) {
+      throw new Error(`Capacity units are not configured when capacity mode is ${CapacityMode.AUTOSCALED}`);
+    }
+    return this._units;
+  }
+
+  public get minCapacity() {
+    if (this._minCapacity === undefined) {
+      throw new Error(`Minimum capacity is not configured when capacity mode is ${CapacityMode.FIXED}`);
+    }
+    return this._minCapacity;
+  }
+
+  public get maxCapacity() {
+    if (this._maxCapacity === undefined) {
+      throw new Error(`Maximum capacity is not configured when capacity mode is ${CapacityMode.FIXED}`);
+    }
+    return this._maxCapacity;
+  }
+
+  public get targetUtilizationPercent() {
+    if (this.mode === CapacityMode.FIXED) {
+      throw new Error(`Target utilization percent is not configured when capacity mode is ${CapacityMode.FIXED}`);
+    }
+    return this._targetUtilizationPercent ?? DEFAULT_TARGET_UTILIZATION;
+  }
+
+  private constructor(mode: string, options: CapacityOptions) {
+    if (options.minCapacity && options.maxCapacity) {
+      if (options.minCapacity > options.maxCapacity) {
+        throw new Error(`Min capacity: ${options.minCapacity} must be less than or equal to max capacity: ${options.maxCapacity}`);
+      }
+    }
+
+    if (options.targetUtilizationPercent && (options.targetUtilizationPercent < 20 || options.targetUtilizationPercent > 90)) {
+      throw new Error(`Target utilization percent must be between 20 and 90, inclusive. Provided: ${options.targetUtilizationPercent}`);
+    }
+
+    this.mode = mode;
+    this._units = options.units;
+    this._minCapacity = options.minCapacity;
+    this._maxCapacity = options.maxCapacity;
+    this._targetUtilizationPercent = options.targetUtilizationPercent;
+  }
+}
+
+/**
+ * Used to configure server-side encryption for a global table and its replicas
+ */
+export class TableEncryptionV2 {
+  /**
+   * Configure table encryption using a DynamoDB owned key.
+   *
+   * Note: This will set the encryption type to AWS_OWNED.
+   */
+  public static dynamoOwnedKey() {
+    return new TableEncryptionV2(TableEncryption.DEFAULT, undefined, undefined);
+  }
+
+  /**
+   * Configure table encryption using an AWS managed key.
+   *
+   * Note: This will set the encryption type to AWS_MANAGED.
+   */
+  public static awsManagedKey() {
+    return new TableEncryptionV2(TableEncryption.AWS_MANAGED, undefined, undefined);
+  }
+
+  /**
+   * Configure table encryption using a customer managed key.
+   *
+   * Note: This will set the encryption type to CUSTOMER_MANAGED.
+   *
+   * @param tableKey the KMS key used for the replica table in the deployment region
+   * @param replicaKeyArns KMS key ARNS for all replica tables except the replica table in the
+   * deployment region
+   */
+  public static customerManagedKey(tableKey: IKey, replicaKeyArns: { [region: string]: string } = {}) {
```

Comment:
If you think we're ever going to need additional options here, better to stick the replica Arns in a struct.